### ISSUE TITLE
Magento 2.3 Compatibility

### ISSUE THREAD

**Issue by robin-homberg** [Flagbit/Magento2-Flysystem#3]
Since Magneto 2.3 will be released soon. Magento2-Flysystem could be tested on the beta of the new Magento version to fix problems with its compatibility to get a Magento 2.3 compatible version of Magento2-Flysystem as soon as possible after the release of the new magento version.

**Comment by flagbird** [comment]
Are all Magento 2 versions supported?

Testing the integration during a build may be of help. This can also be supportive on changes to prevent BC breaks.

**Comment by BrocksiNet** [comment]
Like it is wirtten in [Installation Guide](https://github.com/Flagbit/Magento2-Flysystem/wiki/Installation-&-Configuration-Guide) we support **_Magento2 Version 2.2.x or higher_**. But we should add this also to composer.json. We missed it there @robin-homberg.

But in magento 2.3 there will be maybe breaking changes so we have to test this.

**Comment by robin-homberg** [comment]
While Testing Flysystem on Magento 2.3 there seemed to be an issue inside a core class, which is further described here: https://github.com/magento/magento2/issues/17185.
Still trying to reproduce the not-reproduceability of this issue to get some further hints to fix this (maybe inside the module itself).
There are some more things, which are not working yet on Magento 2.3.x, so the module can't be used for Magento 2.3.x in it's current state. Working on fixes for these issues.

**Comment by flagbird** [comment]
Magento 2.3 is now available. Please keep in mind that PHP 7.2 is now supported too.

**Comment by BrocksiNet** [comment]
Feel free to do this :)

**Comment by flagbird** [comment]
It's assigned to @robin-homberg and it looks like he already knows about some issues. So no one can currently work on this task?

As for myself, I'm already involved in other OS tasks.
You could try a shoutout to the community. Maybe someone wants to contribute. It would be a pity if this project would become unsupported. Every open source project needs to be maintained after it began.

**Comment by robin-homberg** [comment]
Yes. i dont want this project to become unsupported at all. Unfortunately i have currently absolutly no time to continue with this project, while i would really like to :(
I didnt do that much yet about 2.3 compatibility, most things were to find out how to get the magento 2.3 versions work on my local system, because there were multiple bugs on the 2.3 alpha installing process..

One huge issue that needs to be changed for 2.3 is, that magento refactored the Slide-In Container, (f.e. the buttons are rendered completly different now).

**Comment by robin-homberg** [comment]
Update to the Magento 2.3.0 compatibility after some frontend tests:
After Magento 2.3.0 was released i found now some time to do the first testing of the functionality on stable 2.3.0. The most things seem still to work. I realized 2 issues which need to be fixed:
- Like i already mentioned in the testing on one of the magento2.3 alpha branches. Magento refactored the Buttons in the Media Slidein. So in the Flysystem Slidein we should do the same. As a sideeffect of this, the "Open Flysystem Browser" Button is now missing inside the Media Slidein, because of the changed buttons
- The Image Preview is not working anymore. Opening the Image preview will open an empty Lightbox
